### ISSUE TITLE
v2.99.[234]-beta segfaults on opteron 6372

### ISSUE THREAD

**Issue by neuroscr** [xmrig/xmrig#1092]
had to go back to .1

was getting `warning: "can't bind memory"` and then a segfault

Running in LXC container on ubuntu 18LTS, what info do you need? I have 4 sockets, plenty of ram available (256gb in the box, 16gb on the container)

**Comment by xmrig** [comment]
Please run command `./xmrig --export-topology` (for 2.99.3) inside container and if possible outside too then attach resulting `topology.xml` to this issue.

Seems something wrong with hwloc inside container, I will try use LXC container, build miner without hwloc support (`-DWITH_HWLOC=OFF`) should restore 2.99.1 behavior.
Thank you.

**Comment by neuroscr** [comment]
from inside lxc
https://spit.mixtape.moe/view/8e31be8a

**Comment by xmrig** [comment]
I forgot to ask, but what pool do you use? Might be same issue https://github.com/xmrig/xmrig/issues/1066#issuecomment-518080529
Thank you.

**Comment by xmrig** [comment]
According hwloc topology you can use only one (first) CPU (16 cores) (from `0` to `15`), 3 other CPUs not available and miner should not generate config for not exists cores.

* I fixed the crash, but cores above `15` not available anyway.
* You use generated config or custom?

**Comment by neuroscr** [comment]
topology outside https://spit.mixtape.moe/view/7d39e65b

pool: loki.hashvault.pro

I did lock down the LXC container to 16 cpus (because I could no figure out any way to get it to use less CPUs, tried a couple different CLI options).

I did start with an auto config but I did edit it ofc, tried setting up cpu affinity like xmr-stak auto-detected for me:

```
        "rx/loki": [
            0,
            1,
            8,
            9,
            16,
            17,
            24,
            25,
            32,
            33,
            40,
            41,
            48,
            49,
            56,
            57
        ],
```

**Comment by neuroscr** [comment]
also crashes with and without -t

**Comment by xmrig** [comment]
v2.99.5-beta released https://github.com/xmrig/xmrig/releases/tag/v2.99.5-beta

**Comment by neuroscr** [comment]
fixed as of at least 5.10
